### PR TITLE
[RIS-1225] Move to Ruby 3.0.x

### PR DESCRIPTION
**Pull request by trooney-himama**
Ruby `2.7.7` is EOL. Bumping to `3.0.x`.

- Updating minimum Ruby version to Ruby 3.0.5
- Add test support for Ruby 3.1, 3.2, 3.3
- Updated Rubocop to enforce 3.0.x conventions
- Linting fixes

Post-merge:
- Publish `v0.7.0` tag
- Publish to Rubgems

RIS-1225

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -103,3 +103,6 @@ Security/YAMLLoad:
 
 Style/RegexpLiteral:
   Enabled: false
+
+Naming/VariableNumber:
```
Comment (trooney-himama) [inline .rubocop.yml:107]: Disabling this rule as it appears to apply to keys returned from the bambora API.

File: bambora-client.gemspec [hunk 2]
```diff
@@ -46,11 +46,11 @@ Gem::Specification.new do |spec|
...
   spec.add_development_dependency 'bundler', '~> 2'
-  spec.add_development_dependency 'pry', '~> 0.12.0'
-  spec.add_development_dependency 'pry-byebug', '~> 3.7'
+  spec.add_development_dependency 'pry'
+  spec.add_development_dependency 'pry-byebug'
   spec.add_development_dependency 'rake', '~> 13.0'
   spec.add_development_dependency 'rspec', '~> 3.0'
   spec.add_development_dependency 'rspec_junit_formatter', '~> 0.4.1'
-  spec.add_development_dependency 'rubocop', '~> 0.74.0'
+  spec.add_development_dependency 'rubocop', '~> 1.12.0'
```
Comment (trooney-himama) [inline bambora-client.gemspec:54]: Minimum version bumps required to successfully run under Ruby 3.0

File: lib/bambora/adapters/query_string_response.rb [hunk 3]
```diff
@@ -8,7 +8,7 @@ def to_h
       parsed_response = super
       return error_response if parsed_response.values.flatten.empty? # We didn't get a query string back.
 
-      parsed_response.each_with_object({}) { |(key, val), obj| obj[key] = val.length == 1 ? val.first : val }
+      parsed_response.transform_values { |val| val.length == 1 ? val.first : val }
```
Comment (trooney-himama) [inline lib/bambora/adapters/query_string_response.rb:11]: Linting changes from the Rubocop change moving us to 3.0 compatibility. These were all from their "auto-correct" package.

File: lib/bambora/client/version.rb [hunk 4]
```diff
@@ -2,6 +2,6 @@
 
 module Bambora
   class Client
-    VERSION = '0.6.0'
+    VERSION = '0.7.0'
```
Comment (trooney-himama) [inline lib/bambora/client/version.rb:5]: Version bump. Last time I forgot to create a `tag` and release to `ruby-gems` but this time I have the power! ⚔️

File: .editorconfig [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# top-most EditorConfig file
+root = true
+
+# Unix-style newlines with a newline ending every file
+# Indent with 2 spaces
+[*]
+end_of_line = lf
+insert_final_newline = true
+indent_style = space
+indent_size = 2
```
Comment (trooney-himama) [inline .editorconfig:10]: Being a good citizen 🖖

**Review by cristianoventura — DISMISSED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,10 @@
+## 0.7.0 (2024-02-05)
+
+- Updating minimum Ruby version to Ruby 3.0.5
+- Add test support for Ruby 3.1, 3.2, 3.3
+- Updated Rubocop to enforce 3.0.x conventions
+- Linting fixes
+
 ## Unreleased (2023-12-01)
```
Comment (harrylewis) [inline CHANGELOG.md]: Can we merge this with the **Unreleased** items below? Also, can we more explicitly call out breaking change here, of removing support for Ruby `2.7.7`. This library I believe is used by groups that are not just us anymore.

Comment (trooney-himama) [reply CHANGELOG.md]: Updated the wording. Keep in mind downstream clients can leverage `Gemfile` to sort out any issues with our released changes.

**Review by harrylewis — APPROVED** [review]
Thanks for this update, looks good to me. Merge when you are ready!

**Pull request merged**